mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -613,6 +614,12 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<GridwiseGemmOp> {
     LLVM_DEBUG(llvm::dbgs() << "a_block_space: " << a_block_space << "\n");
     LLVM_DEBUG(llvm::dbgs() << "b_block_space: " << b_block_space << "\n");
     LLVM_DEBUG(llvm::dbgs() << "double_block_space: " << block_space << "\n\n");
+
+    // TODO: adjust for data type and device
+    if (block_space * sizeof(float) > 64 * 1024)
```

Comment:
Migrated over here.

Reply:
Ok, so, deeper question: what is the actual limit here?

Reply:
But more immediate note: please file a ticket for that TODO

Reply:
> But more immediate note: please file a ticket for that TODO

Absolutely.

> Ok, so, deeper question: what is the actual limit here?

If I understand it right, the gfx908 limit on lds should be 64k. Therefore the actual limit should account for both data type and size. I guess the only reason we are still fine is due to:
 - Later arch has larger lds size
 - Among supported types, float is the largest 

and since we are bounded by lower bound, this is okay.

Reply:
It actually just reminded me that this scheduling can get complicated in a case like: (Note this may not be a problem today but I can see it is as we fuse more aggresively)
 - We have consumed pretty much all lds space
 - In a fusion pass, it ended up also require some lds space

In each local scope thing are fine because we didn't exceed upper bound of lds. But it will yield a kernel that over allocate and fail at execution time. 

@sjw36 Does fusion do any LDS allocations? If it does we may need some global flag indicating how much space is left globally.

Reply:
Fusion doesn't do LDS allocations